pydantic_ai_slim/pydantic_ai/_pydantic.py

```diff
@@ -76,8 +76,15 @@ def function_schema(  # noqa: C901
     description, field_descriptions = doc_descriptions(function, sig, docstring_format=docstring_format)
 
     if require_parameter_descriptions:
-        if len(field_descriptions) != len(sig.parameters):
-            missing_params = set(sig.parameters) - set(field_descriptions)
+        missing_params = set(sig.parameters) - set(field_descriptions)
+
+        if takes_ctx:
+            for name in missing_params:
+                if _is_call_ctx(sig.parameters[name].annotation):
+                    missing_params.remove(name)
+                    break
+
+        if missing_params:
```

Comment:
What do you think about this shorter implementation? Note that I haven't verified it.

```suggestion
        non_ctx_parameters = set((name for name, param in sig.parameters.items() if not _is_call_ctx(param.annotation)))
        if len(field_descriptions) != len(non_ctx_parameters):
            missing_params = set(non_ctx_parameters) - set(field_descriptions)
```

Reply:
I think that that implementation is a) more complicated and therefore less readable and more difficult to understand which is undesirable in open source codebase imo, and b) incorrect because length comparison will miss discrepancy between parameters listed in docstring and parameters in function definition when number of them is the same, e.g. function is `def foo(a: int):` and docstring lists `:param b: bar param`.